src/Discord/Internal/Rest/Webhook.hs

```diff
@@ -46,16 +45,19 @@ data WebhookRequest a where
   CreateWebhook :: ChannelId -> CreateWebhookOpts -> WebhookRequest Webhook
   GetChannelWebhooks :: ChannelId -> WebhookRequest [Webhook]
   GetGuildWebhooks :: GuildId -> WebhookRequest [Webhook]
-  GetWebhook :: WebhookId -> WebhookRequest Webhook
-  GetWebhookWithToken :: WebhookId -> T.Text -> WebhookRequest Webhook
-  ModifyWebhook :: WebhookId -> ModifyWebhookOpts
+  -- GetWebhook :: WebhookId -> WebhookRequest Webhook
+  GetWebhook :: WebhookId -> Maybe WebhookToken -> WebhookRequest Webhook
+  ModifyWebhook :: WebhookId -> Maybe WebhookToken -> ModifyWebhookOpts
```

Comment:
Nice to add these together, `Maybe` fits well.

From <https://discord.com/developers/docs/resources/webhook#webhook-resource> the advantage of the token is that it doesn't require logging into a bot? But the library requires logging in to send anything at all.

Reply:
I had a thought or two on that as well. the only place the auth is needed is in the restcallthread thing. as in the minimal.hs example you put up on discord a while ago, could we not expand support for webhook only messagin stuff by creating a separate one of those for noauth requests. means that most requests will just bounce, but gives people the option.

Reply:
I think using discord-haskell soley for an authenticated webhook http interface is an edge case. Creating a bot-application is pretty simple and unlocks everything else in the repo. If really desired minimal.hs is up as https://github.com/discord-haskell/discord-haskell/blob/master/examples/rest-without-gateway.hs

Reply:
I'm happy to remove the token cases if we think that they shouldn't be used.

Reply:
Yeah remove the webhook token from Get/Modify/Delete. (sending the token with ModifyWebhook [prevents changing the channelID](https://discord.com/developers/docs/resources/webhook#modify-webhook-with-token) the webhook runs in). 
Execute seems to require it though. 

Change that, and then you should have access to merge the PR